src/x-server.c

```diff
@@ -108,7 +126,17 @@ x_server_get_session_type (DisplayServer *server)
 static gboolean
 x_server_get_can_share (DisplayServer *server)
 {
-    return TRUE;
+    XServerPrivate *priv = x_server_get_instance_private ((XServer*) server);
+    gchar actual_local_hostname[1024];
+
+    g_return_val_if_fail (server != NULL, FALSE);
+
+    if (priv->local_hostname[0] == '\0')
+        return TRUE;
+
+    gethostname (actual_local_hostname, 1024);
```

Comment:
Add a comment above here about why the hostname affects the ability to share this server.

Reply:
done :)